executable-spec/test/Cardano/Ledger/Spec/STS/Chain/Chain/Properties.hs

```diff
@@ -46,3 +53,243 @@ qc_revealsAreClassified
       & dom
       & length
       & fromIntegral
+
+-- | Test coverage describes the test space that we are targeting for.
+-- Describe the most essential characteristics of our generated traces.
+--
+-- TODO: this must be updated to reflect all phases in the lifecycle of
+-- a software update and not just the Ideation phase.
+relevantCasesAreCovered :: QC.Property
+relevantCasesAreCovered
+  = QC.withMaxSuccess 300
+  $ STS.Gen.forAllTrace @(CHAIN ShortHash) @() maxTraceLength ()
+  $ \traceSample ->
+      -- traces should be long enough to allow for the stabilization
+      -- of events
+      QC.cover 80
+        ( (fromIntegral $ Trace.traceLength traceSample)
+          >=
+          getMinTraceLength  (Chain.k $ Trace._traceEnv traceSample)
+        )
+        "Trace length is long enough to allow for the stabilization of events"
+        $
+
+      -- 80% of traces should include a 20% percent of update payload
+      QC.cover 80
+        ( (updatePayloadPct traceSample) >= 15
+        &&
+          (updatePayloadPct traceSample) <= 25
+        )
+        "a reasonable pct of update payload appears in the trace "
+        $
+
+      -- Lifecycle coverage:
+      -- There is at least one proposal in every phase of the lifecycle
+      QC.cover 50
+        (lifecycleCoverage traceSample)
+        "The lifecycle of a software update is sufficently covered"
+        $
+
+      -- X% of traces should: there are SIPs that got approved
```

Comment:
Maybe `outcomeAny` should be renamed to something more descriptive like `lastStateContainsOutcome`?

We could even use such function infix if you like:

```haskell
traceSample `lastStateContainsOutcome` Data.Approved
```

Reply:
OK I will change it.

Reply:
Done. OK.